core/io/marshalls.cpp

```diff
@@ -52,9 +52,9 @@ ObjectID EncodedObjectAsID::get_object_id() const {
 	return id;
 }
 
-#define _S(a) ((int32_t)a)
-#define ERR_FAIL_ADD_OF(a, b, err) ERR_FAIL_COND_V(_S(b) < 0 || _S(a) < 0 || _S(a) > INT_MAX - _S(b), err)
-#define ERR_FAIL_MUL_OF(a, b, err) ERR_FAIL_COND_V(_S(a) < 0 || _S(b) <= 0 || _S(a) > INT_MAX / _S(b), err)
+#define S(a) ((int32_t)a)
```

Comment:
That's a pretty bad name IMO which can potentially conflict with even more code than `_S` would.

I would give it a more explicit name (`INT_S` maybe?) and `#undef` it after use. Or just put `int32_t` casts directly inline instead of using the macro.

CC @Faless

Reply:
Yeah, we can use `(int32_t)` directly, but we need to be careful with parenthesis. This should be always safe (even safe than what we have now?):
```
((int32_t)(a))
```

Reply:
I ended up inlining the cast.  Those two lines are now a bit longer now tho ^^"